lib/registry.js

```diff
@@ -14,9 +13,62 @@ var BaseElement;
  */
 
 var ElementRegistry = createClass({
-  constructor: function() {
+  constructor: function(noDefault) {
     this.elementMap = {};
     this.elementDetection = [];
+    this.BaseElement = require('./primitives/base-element')(this);
```

Comment:
One thing that could be done would be:

```js
var ElementRegistry = createClass({
  constructor: function(options) {
    // ...
    this.BaseElement = require('./primitives/base-element')(this, options.classMethods, options.staticMethods, options.getterSetters);
    // ...
  }
  
  // ....
});
```

> A namespace module may need to iterate over all currently registered elements and adjust their prototypes accordingly

I could see this as an option for people was well.